src/reducers/app-status.reducer.js

```diff
@@ -2,32 +2,55 @@
 import {
   START_DELETING_PROJECT,
   FINISH_DELETING_PROJECT,
+  REINSTALL_DEPENDENCIES_START,
+  REINSTALL_DEPENDENCIES_FINISH,
+  SET_STATUS_TEXT,
+  RESET_STATUS_TEXT,
   DELETE_PROJECT_ERROR,
 } from '../actions';
 
 import type { Action } from 'redux';
 
 type State = {
   blockingActionActive: boolean,
+  statusText: string,
 };
 
-const initialState = {
+export const initialState = {
   blockingActionActive: false,
+  statusText: 'Please wait...',
 };
 
 export default (state: State = initialState, action: Action = {}) => {
   switch (action.type) {
     case START_DELETING_PROJECT:
+    case REINSTALL_DEPENDENCIES_START:
       return {
+        ...state,
         blockingActionActive: true,
       };
 
     case FINISH_DELETING_PROJECT:
+    case REINSTALL_DEPENDENCIES_FINISH:
     case DELETE_PROJECT_ERROR:
       return {
+        ...state,
         blockingActionActive: false,
       };
 
+    case SET_STATUS_TEXT:
+      const newStatus = action.statusText;
+      return {
+        ...state,
+        statusText: newStatus !== '' ? newStatus : state.statusText, // only update if string defined
+      };
```

Comment:
We probably want to avoid creating a new object if we don't have anything to change. So we could do:

```js
if (newStatus === '') {
  return state;
}

return {
  ...state,
  statusText: newStatus,
}
```

The rationale is that redux reducers should return the existing object when nothing changes, to `PureComponent`s won't re-render when they don't need to.

Reply:
Oh, good point. I'll change that.